src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Services/DataSetVersionService.cs

```diff
@@ -123,4 +142,264 @@ private void DeleteParquetFiles(DataSetVersion dataSetVersion)
+    }
+
+    private async Task<Either<ActionResult, ReleaseFile>> GetReleaseFile(
+        Guid releaseFileId,
+        CancellationToken cancellationToken)
+    {
+        var releaseFile = await contentDbContext.ReleaseFiles
+            .Include(rf => rf.File)
+            .Include(rf => rf.ReleaseVersion)
+            .FirstOrDefaultAsync(rf => rf.Id == releaseFileId, cancellationToken);
+
+        return releaseFile is null
+            ? ValidationUtils.ValidationResult(CreateReleaseFileIdError(
+                message: ValidationMessages.FileNotFound,
+                releaseFileId: releaseFileId
+            ))
+            : releaseFile;
+    }
+
+    private async Task<Either<ActionResult, Unit>> ValidateReleaseFile(
+        ReleaseFile releaseFile,
+        CancellationToken cancellationToken)
+    {
+        // ReleaseFile must not already have a DataSetVersion
+        if (await publicDataDbContext.DataSetVersions.AnyAsync(
+                dsv => dsv.ReleaseFileId == releaseFile.Id,
+                cancellationToken: cancellationToken))
+        {
+            return ValidationUtils.ValidationResult(
+            [
+                CreateReleaseFileIdError(
+                    message: ValidationMessages.FileHasApiDataSetVersion,
+                    releaseFileId: releaseFile.Id)
+            ]);
+        }
+
+        // ReleaseFile must relate to a ReleaseVersion in Draft approval status
+        if (releaseFile.ReleaseVersion.ApprovalStatus != ReleaseApprovalStatus.Draft)
+        {
+            return ValidationUtils.ValidationResult(
+            [
+                CreateReleaseFileIdError(
+                    message: ValidationMessages.FileReleaseVersionNotDraft,
+                    releaseFileId: releaseFile.Id)
+            ]);
+        }
+
+        List<ErrorViewModel> errors = [];
+
+        // ReleaseFile must relate to a File of type Data
+        if (releaseFile.File.Type != FileType.Data)
+        {
+            errors.Add(CreateReleaseFileIdError(
+                message: ValidationMessages.FileTypeNotData,
+                releaseFileId: releaseFile.Id));
+        }
+
+        // There must be a ReleaseFile related to the same ReleaseVersion and Subject with File of type Metadata
+        if (!await contentDbContext.ReleaseFiles
+                .Where(rf => rf.ReleaseVersionId == releaseFile.ReleaseVersionId)
+                .Where(rf => rf.File.SubjectId == releaseFile.File.SubjectId)
+                .Where(rf => rf.File.Type == FileType.Metadata)
+                .AnyAsync(cancellationToken: cancellationToken))
+        {
+            errors.Add(CreateReleaseFileIdError(
+                message: ValidationMessages.NoMetadataFile,
+                releaseFileId: releaseFile.Id));
+        }
+
+        return errors.Count == 0 ? Unit.Instance : ValidationUtils.ValidationResult(errors);
+    }
+    
+    private async Task<Either<ActionResult, Unit>> ValidateReleaseFileAndDataSet(
```

Comment:
Bit confusing to have another validation method as it looks like we could just combine this all into `ValidateReleaseFile`?

Would suggest we merge these two methods together.